### ISSUE TITLE
Results

### ISSUE THREAD

**Issue by publishwhatyoufund** [pwyf/2022-Index-indicator-definitions#34]
## Description
The results show whether activities achieved their intended outputs in accordance with the stated goals or plans.

This is only expected if the activity is in the implementation, completion or post-completion phase.

This is not expected if the default aid type code is administrative costs (G01).

## Proposed tests

Firstly (results data):
```python
   Given an IATI activity
    And the activity is current
    And `activity-status/@code` is one of 2, 3 or 4
    And `default-aid-type/@code` is not G01
    Then `result` should be present
 ```
Secondly (results documents):
```python
   Given an IATI activity
    And the activity is current
    And `activity-status/@code` is one of 2, 3 or 4
    And `default-aid-type/@code` is not G01
    Then `document-link/category[@code="A08"]` should be present
```

**Comment by hsd25** [comment]
Hey PWYF team, when talking about results, what is expected? indicators and their targets, or also actual results? if actual results is expected, what is the rule? Thanks a lot.

**Comment by ToonVB** [comment]
Hello, in practice there is no info on (intended nor achieved) results for core contributions either (type of aid B01-B04), where it is the implementer that owns & publishes this info. For small scale type of aid D & H activities (of which bilateral donors have a lot of ad hoc contributions) this info is regularly lacking as well. Finally, the same goes for type of aid F financing arrangements.
In short , as is the case with a number of other info-elements, this is only collected/available in a systematical way for type of aid C1 activities = specific projects.

**Comment by breidertmt** [comment]
How does this treat for activities that have just started and might not have results. Typically, for the US grant and contracting cycle there is a quarterly reporting which happens 45-90 days after the close of a quarter.

**Comment by stevieflow** [comment]
**1. _Actual_ results data is in the indicator element**

The IATI standard has a ``result`` element - which this test looks for.  

However, in order to _actually_ declare data, a publisher has to then include child elements.  

For example:

- ``'result/indicator``
- ``result/indicator/period/target``
- ``result/indicator/period/actual``

Therefore, the test is false in terms of "stopping" at the ``result`` element, as it includes no function to check for actual results data.

**2. How are up to date results evaluated ?**

The test has no information in terms of how the currency of results are evaluated.  Page 52 of the Methodology states:

> If the activity is ongoing then up to date results should be available

How would "up to date results" be tested for?  Are these relevant to the start and end dates of the project, for example? The test offers no guidance on this.

**3. No test for closed activities**

The Methodology states:

> If the activity has ended then the actual results should be available within 12-months of the ending

- This is not reflected in the above test (the test only considers current activities)
- Actual results would be in ``result/indicator/period/actual`` - which is not stated in the test
 
**4. Qualitative results**

An indicator can have a "measure type", whereas it is declared as a unit or percentage, for example: https://codelists.codeforiati.org/IndicatorMeasure/.  Also included in this codelist is "qualitative".  This would be of a narrative nature rather than numeric, and may not always fit well within the IATI schema.   Neither the test or methodology have any concern for the type of data that might be published

NOTE: a side issue on this is that d-portal does not currently display the written ``comment`` elements of any result/indicator.  This might be important context if @publishwhatyoufund reviewers are using d-portal (which is very helpful in many many other aspects): https://github.com/devinit/D-Portal/issues/609


**5. Differing publishing practices in IATI** 

It should be noted that IATI publishers have differing approaches to how they might describe the target and actual data, which can lead to differing (sic!) results!  We might expect the target and actual to be in a "pair" in the XML, but it is also common practice that a target can have a blank actual in one period, and vice versa in another.  There has to be some pragmatism in terms of how the data is assessed, as there is no common best practice in the IATI community, presently.  The test and the methodology have nothing to say about this - partly because the test stops abruptly at the ``result`` element.

**Comment by AndieVaughn** [comment]
Some USAID activities by default will never have activity results to publish. We propose excluding A01, B02, and B04 aid types or something similar to hone the test’s measurement of transparency. We proposed a similar change in the last review of methodology and maintain the results for core contributions, pooled funds, etc. cannot be accurately reported by donors who cannot track specific funds after disbursement.

**Comment by ForeignCommonwealthDevelopmentOffice** [comment]
FCDO agrees with comments above from Andie, breidertmt and Toon, as results are included in FCDO annual reviews and our logframes, but this is focused on programme activities.  Some of the other elements in the code lists are less likely to have a logframe because results are owned by others eg. some core funded organisations rather than centrally.

**Comment by jzheng-79** [comment]
As there are major changes in 'Results' between IATI schema 2.02 & 2.03, we would like to make sure that this test will be correctly applied for publishers using schema 2.02 as well.

**Comment by stevieflow** [comment]
> As there are major changes in 'Results' between IATI schema 2.02 & 2.03

for info / context - the changelog for 2.02 to 20.3 can be found at https://iatistandard.org/en/iati-standard/upgrades/upgrade-changelogs/decimal-upgrade-to-2-03/2-03-changes/#activities-schema

**Comment by publishwhatyoufund** [comment]
**@hsd25, @breidertmt**: for the results indicator, results data or a results document are accepted. These should include results against expected results/targets and can include log frames or results chains. Both current and completed activities are considered for this indicator. If the activity is ongoing then actual results should be available. If the activity has started recently (within the last 18 months), then no actual results are expected. If the activity has closed, then final results are expected within 12 months of the end date. 

**@stevieflow**: given the different ways publishers publish their results we have kept this test broad. Up to date and final results are tested in our manual sampling – see response to @hsd25 for the parameters used to assess these.
Either quantitative or qualitative results are accepted for this indicator.
We review the results data in our IATI sampling and check for consistency of publication of results.

**@ToonVB, @AndieVaughn, @ForeignCommonwealthDevelopmentOffice**: when this was raised in the previous consultation we surveyed some of the data published for core contributions and found that some organisations are publishing results for these. Aid types D and H including Technical Assistance, public awareness raising, and in-donor refugee costs can also have results. On this basis we decided not to exclude these aid types. We will exclude aid type F – debt relief. 

**@jzheng-79, @stevieflow:** thank you for flagging the changes in the schema, we will make sure these are accounted for in the test and that the test works correctly on both versions 2.02 and 2.03.
